### ISSUE TITLE
Train/Validation/Test datasets

### ISSUE THREAD

**Issue by junxnone** [junxnone/AI#6]
# Reference
- [ ] [测试数据集（Test Set) 和验证数据集 (Validation Set) 之间的区别](https://zhuanlan.zhihu.com/p/29133576)
- [ ] [Keras中的validation data并不用于更新权重](https://blog.csdn.net/weixin_34186128/article/details/87343645)
- [ ] [keras issue discuss](https://github.com/keras-team/keras/issues/1753)
- [ ] [几种交叉验证（cross validation）方式的比较](https://www.cnblogs.com/ysugyl/p/8707887.html)

# Train dataset
是一些我们已经知道输入和输出的数据集训练机器去学习，通过拟合去寻找模型的初始参数。例如在神经网络（Neural Networks)中， 我们用训练数据集和反向传播算法（Backpropagation）去每个神经元找到最优的比重（Weights)。

# Validation dataset
也是一些我们已经知道输入和输出的数据集，通过让机器学习去优化调整模型的参数，在神经网络中， 我们用验证数据集去寻找最优的网络深度（number of hidden layers)，或者决定反向传播算法的停止点；在普通的机器学习中常用的交叉验证（Cross Validation) 就是把训练数据集本身再细分成不同的验证数据集去训练模型。
- `keras` 中的 `validation data` 不会用于更新权重

# Test dataset

用户测试模型表现的数据集，根据误差（一般为预测输出与实际输出的不同）来判断一个模型的好坏。

# 数量分布

在传统的机器学习中，这三者一般的比例为training/validation/test = 50/25/25, 但是有些时候如果模型不需要很多调整只要拟合就可时，或者training本身就是training+validation (比如cross validation)时，也可以training/test =7/3.
- 2:1:1
- 8:2:5
- 7:3

# Eyeball(Ear) & Blackbox

- Eyeball(Ear) : 在 Eyeball 开发集中建立了对样本的直观认识
> 对错误样本进行误差分析
>> **有约 50 个错误样本，你将能够比较好地了解主要的误差来源**
> 分类器的错误率越低，为了获得足够多的错误样本进行误差分析，需要的 Eyeball 开发集就越大。

> Eyeball 开发集的性能比 Blackbox 开发集提升得更快，说明已经过拟合到 Eyeball 开发集了。此时可能需要丢弃它并寻找一个新的 Eyeball 开发集，比如可以将更多 Blackbox 开发集中的样本移到 Eyeball 开发集中，也可以考虑去获取新的标注数据。
- Blackbox: 通过测量错误率来自动评估分类器，也可以使用它来选择算法或调整超参数
